core/src/main/java/com/amplifyframework/auth/options/AuthSignUpOptions.java

```diff
@@ -18,60 +18,55 @@
 import androidx.annotation.NonNull;
 import androidx.core.util.ObjectsCompat;
 
+import com.amplifyframework.auth.AuthUserAttribute;
 import com.amplifyframework.util.Immutable;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
-public final class AuthSignUpOptions {
-    private final Map<String, String> userAttributes;
-    private final Map<String, String> validationData;
+public class AuthSignUpOptions {
+    private final List<AuthUserAttribute> userAttributes;
 
     /**
      * Advanced options for signing in.
      * @param userAttributes Additional user attributes which should be associated with this user on registration
-     * @param validationData A map of custom key/values to be sent as part of the sign up process
      */
-    private AuthSignUpOptions(
-            Map<String, String> userAttributes,
-            Map<String, String> validationData
+    protected AuthSignUpOptions(
+            List<AuthUserAttribute> userAttributes
     ) {
         this.userAttributes = userAttributes;
-        this.validationData = validationData;
     }
 
     /**
      * Get additional user attributes which should be associated with this user on registration.
      * @return additional user attributes which should be associated with this user on registration
      */
     @NonNull
-    public Map<String, String> getUserAttributes() {
+    public List<AuthUserAttribute> getUserAttributes() {
         return userAttributes;
     }
 
-    /**
-     * Get a map of custom key/values to be sent as part of the sign up process.
-     * @return a map of custom key/values to be sent as part of the sign up process
-     */
-    @NonNull
-    public Map<String, String> getValidationData() {
-        return validationData;
-    }
-
     @NonNull
-    public static Builder builder() {
-        return new Builder();
+    public static Builder<?> builder() {
+        return new CoreBuilder();
     }
 
+    /**
+     * When overriding, be sure to include the parent properties in the hash.
+     * @return Hash code of this object
+     */
     @Override
     public int hashCode() {
         return ObjectsCompat.hash(
-                getUserAttributes(),
-                getValidationData()
+                getUserAttributes()
         );
     }
 
+    /**
+     * When overriding, be sure to include the parent properties in the comparison.
```

Comment:
In your base classes, you could probably do like:
```
if (!super.equals(obj))
```
But, you'll have to remove the class-type checks. You probably want to relax them to `(obj instanceof BaseType)` instead of "obj _is_ basetype"?

Reply:
ok

Reply:
If the child class has properties set which are not present in the base class, the two are not equal so combined with my explanatory comment above, I don't think this is the right path to go.